Testscripts/Windows/FCOPY-NEGATIVE.ps1

```diff
@@ -124,90 +124,104 @@ function Main {
+    if ($? -eq $true) {
+        Write-LogInfo "File has been copied to guest VM when 'Guest Service Interface' enabled"
+        # Clean the /tmp directory
+        Run-LinuxCmd -username $VMUserName -password $VMPassword -ip $ipv4 -port $VMPort "rm -rf /tmp/*" -runAsSudo
+    } else {
+        Write-LogErr  "Fail to copy $filePath to guest VM even 'Guest Service Interface' enabled"
+        return "FAIL"
     }
 
     #
-    # Step 2: verify the file cannot copy to vm when "Guest Service Interface" is disabled
+    # Step 3: verify the file cannot copy to vm when "Guest Service Interface" is disabled
     #
-    Write-LogInfo "Info: Step 2: fcopy file to vm when 'Guest Service Interface' is disabled"
+    Write-LogInfo "Info: Step 3: fcopy file to vm when 'Guest Service Interface' is disabled"
     Disable-VMIntegrationService -Name "Guest Service Interface" -vmName $VMName -ComputerName $HvServer
-    if ( $? -eq $false) {
+    if ($? -eq $false) {
         Write-LogErr "Fail to disable 'Guest Service Interface'"
         return "FAIL"
     }
 
-    $Error.Clear()
-    Copy-VMFile -vmName $VMName -ComputerName $HvServer -SourcePath $filePath -DestinationPath "/tmp/" -FileSource host -ErrorAction SilentlyContinue
-
-    if ( $? -eq $true ) {
+    Copy-VMFile -vmName $VMName -ComputerName $HvServer -SourcePath $filePath `
+        -DestinationPath "/tmp/" -FileSource host -ErrorAction SilentlyContinue
+    if ($? -eq $true) {
         Write-LogErr "File has been copied to guest VM even 'Guest Service Interface' disabled"
         return "FAIL"
     }
-    elseif (($Error.Count -gt 0) -and ($Error[0].Exception.Message -like "*FAIL to initiate copying files to the guest*")) {
-        Write-LogInfo "Info: File could not be copied to VM as expected since 'Guest Service Interface' disabled"
 
-        #
-        # Step 3: verify the file cannot copy to vm when hypervfcopyd is stopped
-        #
-        Write-LogInfo "Info: Step 3: fcopy file to vm when hypervfcopyd stopped"
-        Enable-VMIntegrationService -Name "Guest Service Interface" -vmName $VMName -ComputerName $HvServer
-        if ( $? -ne $true) {
-            Write-LogErr "Fail to enable 'Guest Service Interface'"
-            return "FAIL"
-        }
+    Write-LogInfo "Info: File could not be copied to VM as expected since 'Guest Service Interface' disabled"
 
-        # Stop fcopy daemon to do negative test
-        $sts = Stop-FcopyDaemon -VmPassword $VMPassword -vmPort $VMPort -vmUserName $VMUserName -ipv4 $Ipv4
-        if (-not $sts[-1]) {
-            Write-LogErr "FAIL to stop hypervfcopyd inside the VM!"
-            return "FAIL"
-        }
+    #
+    # Step 4: verify the file cannot copy to vm when hypervfcopyd is stopped
+    #
+    Write-LogInfo "Info: Step 4: fcopy file to vm when hypervfcopyd stopped"
+    Enable-VMIntegrationService -Name "Guest Service Interface" -vmName $VMName -ComputerName $HvServer
```

Comment:
There is a chance the service name change. Please replace them with the global constant in the file. So we will update one place with new service name in future.

Reply:
Add a new global variable named VMIntegrationGuestService.